src/routes/tags.rs

```diff
@@ -20,17 +20,12 @@ pub fn config(cfg: &mut web::ServiceConfig) {
     );
 }
 
-// TODO: searching / filtering? Could be used to implement a live
-// searching category list
 #[get("category")]
 pub async fn category_list(pool: web::Data<PgPool>) -> Result<HttpResponse, ApiError> {
     let results = Category::list(&**pool).await?;
     Ok(HttpResponse::Ok().json(results))
 }
 
-// At some point this may take more info, but it should be able to
-// remain idempotent
-// TODO: don't fail if category already exists
```

Comment:
This still fails if the category already exists.  This should also still have more info, such as a category description.

Reply:
Isn't that the point? What should it do if the category exists

Reply:
This method should act the same as when a category was created.  Either return a 200 or 204 status code, and do nothing.